### PR TITLE
Fix myid user/group perm and improve zookeeper_servers variable handling

### PR DESCRIPTION
**Pull request by gothicx**
I have also moved the logs from the installation folder to /var/log. This will be better when you have a LVM partition just for /var/log, so when it gets full, it doesn't break anything.

### CODE REVIEW

**Comment by sleighzy** [comment]
Hi @gothicx , thanks for the PR. Just finished running it a few times this evening so will drop some comments on it.

**Comment by sleighzy** [comment]
Can you please run `ansible-lint **/*.yaml` over this as there are some linting issues that need to be resolved.

File: tasks/main.yaml
Comment (sleighzy) [inline tasks/main.yaml]: Linting issue: "Don't compare to literal True/False".

Similar change required in other lines.

File: tasks/main.yaml
Comment (sleighzy) [inline tasks/main.yaml]: This breaks the idempotent nature of this. Each run is deleting this dir and then recreating the symlink further down so always comes up with 2 changes. Is this task to clean up from previous versions of this role prior to your changes where this `/usr/share/zookeeper/log` directory would have existed previously? This would also mean that all application logs would be deleted for any existing installation where this was run. I agree that `/var/log/zookeeper` is preferable, just wanting to consider options for mitigating this.

**Review by sleighzy — CHANGES_REQUESTED** [review]

File: tasks/main.yaml
Comment (sleighzy) [reply tasks/main.yaml]: Apologies, please ignore the second part of this comment. This role didn't used to create a `/usr/share/zookeeper/log` directory previously so there shouldn't be any loss of data. The first part of the comment should still be accurate however, this is deleting and re-adding the directory symlink so is not idempotent.

**Comment by sleighzy** [comment]
Hi @gothicx , let me know if you have further changes coming. I have another PR for adding linting checks with a GitHub Action. That PR also makes similar changes for the user/group perms and directory modes. I can hold off for a while until yours is completed so there's no merge conflicts for you.

File: tasks/main.yaml
Comment (gothicx) [reply tasks/main.yaml]: How should we handle this? As the directory logs inside the unpacked zookeeper will always exists.

Comment (sleighzy) [reply tasks/main.yaml]: I'm a little confused now. The role already creates the `/var/log/zookeeper` directory and this is set as the value in the `ZOO_LOG_DIR` environment variable value, https://github.com/sleighzy/ansible-zookeeper/blob/master/templates/default.j2#L1, which ZooKeeper uses for it's logs when it starts up. There shouldn't be any log dirs in the unpacked ZooKeeper installation directory. If you are wanting to have a symlink to the `/var/log/zookeeper` dir under `/usr/share/zookeeper` dir then you can leave that single task there for just adding the link.

Comment (gothicx) [reply tasks/main.yaml]: You're right, there is no logs folder inside the tarball. I must have made some previous tests and it did create one and assumed that wrongly.

**Comment by gothicx** [comment]
This should be good now for merging. Thank you.

**Comment by sleighzy** [comment]
LGTM. Thanks for that.

**Review by sleighzy — APPROVED** [review]

**Pull request merged**
